0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -110,508 +115,638 @@ informative:
+publicKey
+: is the DER encoded RSA public key 
 
 signature
-: is an octet string representing the RSA signature. It MUST be encoded in big-endian byte order with the exact same number of octets as the modulus, even if this means adding leading zeros. This ensures that the fulfillment size is constant and known ahead of time. Note that the field is still binary encoded with a length prefix for consistency.
+: is an octet string representing the RSA signature. It MUST be encoded in big-endian byte order. with the exact same number of octets as the modulus of the public key, even if this means adding leading zeros. This ensures that the fulfillment size is constant and known ahead of time. Note that the field is still binary encoded with a length prefix for consistency.
 
 : Implementations MUST verify that the signature and modulus consist of the same number of octets and that the signature is numerically less than the modulus.
 
 The message to be signed is provided separately. If no message is provided, the message is assumed to be an octet string of length zero.
 
 ### Implementation {#rsa-sha-256-condition-type-implementation}
-The recommended modulus size as of 2016 is 2048 bits [KEYLENGTH-RECOMMENDATION](#KEYLENGTH-RECOMMENDATION) . In the future we anticipate an upgrade to 3072 bits which provides approximately 128 bits of security [NIST-KEYMANAGEMENT](#NIST-KEYMANAGEMENT) (p. 64), about the same level as SHA-256.
+The recommended modulus size as of 2016 is 2048 bits [KEYLENGTH-RECOMMENDATION](#KEYLENGTH-RECOMMENDATION). In the future we anticipate an upgrade to 3072 bits which provides approximately 128 bits of security [NIST-KEYMANAGEMENT](#NIST-KEYMANAGEMENT) (p. 64), about the same level as SHA-256.
 
-### Example {#rsa-sha-256-example}
-
-Example condition:
+### Validating {#rsa-sha-256-condition-type-validating}
 
-    cc:3:11:Bw-r77AGqSCL0huuMQYj3KW0Jh67Fpayeq9h_4UJctg:260
+An RSA-SHA-256 fulfillment is valid iff :
+ 
+  1. F.signature is valid for the message M, given the RSA public key F.publicKey.
+  2. D is equal to C. 
 
-Example fulfillment:
+### Example {#rsa-sha-256-example}
 
-    cf:3:gYCzDnqTh4O6v4NoUP9J4U-H4_ktXEbjP-yj5PCyI1hYCxF2WZX0uO6n-0cSwuHjFvf3dalT0jIhahadmmTdwAcSCkALN_KvwHe2L-ME3nTeahGexAdrUpxPYJawuq1PUz3wFzubgi_YXWX6S--pLY9ST2nLygE2vYDQlcFprsDglYGAjQM0-Z5B-953uQtJ5dXL1D5TWpM0s0eFF0Zty7J2Y3Nb0PqsR5I47a2wYlA7-106vjC8gHFdHVeSR6JksSrhj8YaMWfV0A6qhPz6hq-TqSKCXd4mf3eCpyyFYR_EyH5zXd56sJEU3snWlFbB_bKAW4si_qdfY9dT87YGUp_Grm0
+TODO
 
+## ED25519 {#ed25519-condition-type}
+ED25519 is assigned the type ID 4. It relies on the SHA-512 digest algorithm and the ED25519 signature scheme as the condition fingerprint is not a digest.
 
+The exact algorithm and encodings used for the public key and signature are defined in [I-D.irtf-cfrg-eddsa](#I-D.irtf-cfrg-eddsa) as Ed25519. SHA-512 is used as the hashing function for this signature scheme.
```

Comment:
Note that this draft should shortly be published as RFC 8032 (presently in IETF AUTH48 state, approved by one author but waiting for approval from another):

https://www.rfc-editor.org/auth48/rfc8032https://www.rfc-editor.org/auth48/rfc8032